### ISSUE TITLE
Bulid Failing with Latest Amplify CLI

### ISSUE THREAD

**Issue by jeffbking** [aws-amplify/amplify-cli#9170]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v14.17.5

### Amplify CLI Version

6.3.1

### What operating system are you using?

Mac

### Amplify Categories

Not applicable

### Amplify Commands

Not applicable

### Describe the bug

Build is only failing in the cloud CI/CD. No changes were made to my code base. Building/publishing locally works using 6.3.1 that is on my machine. 

The build in CI/CD worked yesterday (Nov 31), and I just re-triggered the same branch to build through amplify console and it fails to update Lambdas with:

```
UPDATE_FAILED               functionXXXX        AWS::CloudFormation::Stack Wed Dec 01 2021 16:03:08 GMT+0000 (Coordinated Universal Time) Parameters: [s3Key, deploymentBucketName] must have values
UPDATE_FAILED               functionYYYY        AWS::CloudFormation::Stack Wed Dec 01 2021 16:03:08 GMT+0000 (Coordinated Universal Time) Parameters: [s3Key, deploymentBucketName] must have values
UPDATE_FAILED               functionZZZZ           AWS::CloudFormation::Stack Wed Dec 01 2021 16:03:08 GMT+0000 (Coordinated Universal Time) Parameters: [s3Key, deploymentBucketName] must have values
```

### Expected behavior

It should build properly.

### Reproduction steps

Trigger a build in the CI/CD system through the amplify console and it no longer works.

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by jeffbking** [comment]
Confirmed working in CI/CD if I pin version 7.5.4 of Amplify CLI in the Amplify portal build settings. The issue seems to have been introduced in 7.5.5

**Comment by lazpavel** [comment]
Hi @jeffbking, thank you for reporting this issue. The team is actively looking into it.

**Comment by hisham** [comment]
I'm getting this error as well.

**Comment by hisham** [comment]
On the 2nd push it worked though and I didn't change amplify version.

**Comment by KalvinWei** [comment]
Faced the same issue.
Recently, I changed nothing to my Amplify backend. 
Last deployment was successful but today's failed and stopped at backend deployment.

**Comment by nunesky** [comment]
Hi, we have the same issue here. Our Amplify version is 6.1.1
There was no change on the Amplify backend, but the build is breaking with the same message. Our first failure was reported on Wed Dec 01 2021 11:35:37 GMT+0000, if that helps. Thx!

**Comment by KalvinWei** [comment]
I solved the problem by reverting my CLI to v7.5.4 and set the Amplify build CLI version from `latest` to `7.5.4`
